src/widget/container/mod.rs

```diff
@@ -60,28 +64,45 @@ where
         Self { options }
     }
 
+    fn main_axis(&self) -> Axis {
+        self.options.axis
+    }
+
+    fn cross_axis(&self) -> Axis {
+        self.options.axis.opposite()
+    }
+
     fn content_size(&self) -> IntrinsicSize {
         self.options
             .children
             .iter()
             .fold(IntrinsicSize::none(), |size, widget| {
                 let widget_size = widget.intrinsic_size();
 
-                let width = match (size.width, widget_size.width) {
-                    (Some(width), Some(widget_width)) => Some(width.max(widget_width)),
-                    (Some(width), None) => Some(width),
-                    (None, Some(widget_width)) => Some(widget_width),
+                let cross_axis_size = match (
+                    size.for_axis(self.cross_axis()),
+                    widget_size.for_axis(self.cross_axis()),
+                ) {
+                    (Some(size), Some(widget_size)) => Some(size.max(widget_size)),
+                    (Some(size), None) => Some(size),
+                    (None, Some(widget_size)) => Some(widget_size),
                     (None, None) => None,
```

Comment:
This is pretty much unchanged other than rewording it to make sense for either dimension.